woocommerce/woocommerce
# [e2e tests] Check email for completed order

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes [/woocommerce/woocommerce-quality/issues/656](https://github.com/woocommerce/woocommerce-quality/issues/656)

This updates order-emails with a test to verify that a completed order email was received and its content. The plug-in WP Mail Logging is used along these tests too.



### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm test:e2e-pw ./tests/e2e-pw/tests/merchant/order-emails.spec.js --headed` and verify that tests pass.
2. Code review and verify approach.
3. Make sure that CI jobs are ✅

### Changelog entry




<details>

#### Significance




#### Type




#### Message

This aims to add more checks to confirm that a completed order email is received, as well as its content.

#### Comment

</details>
